src/ngraph/op/batch_norm.hpp

```diff
@@ -36,6 +36,15 @@ namespace ngraph
                       std::shared_ptr<Node> beta,
                       std::shared_ptr<Node> input);
 
+            // BatchNorm Training
```

Comment:
IMO it would be helpful to increase the header comments regarding each of these constructors.

For example, it would be helpful if each constructor's comments indicated what the output type of this op is when using that particular c'tor.

Reply:
I second the suggestion to have separate classes. It would simplify constructors, generate_adjoints and even emitter handlers. This is getting really hard to follow.